internal/csi-addons/networkfence/fencing.go

```diff
@@ -287,6 +287,57 @@ func (nf *NetworkFence) AddClientEviction(ctx context.Context) error {
 	return nil
 }
 
+func getBlocklistEntries(ctx context.Context) ([]string, error) {
+	// FIXME: replace the ceph command with go-ceph API in future
+	cmd := []string{"osd", "blocklist", "ls"}
+	stdout, stdErr, err := util.ExecCommandWithTimeout(ctx, 5*time.Minute, "ceph", cmd...)
+	if err != nil {
+		return nil, fmt.Errorf("failed to list blocklisted clients: %w, stderr: %q", err, stdErr)
+	}
+
+	// Parse the stdout to extract blocklist entries
+	// each entry is a separate line in the output
+	entries := strings.Split(strings.TrimSpace(stdout), "\n")
```

Comment:
Give an example of the output that is expected from the ceph command. It is difficult to review this without knowing what the command returns.

Even better: move the parsing to a helper function and add a unit-test.

Reply:
Thanks Niels, have added comments, explaining the workflow here and in main removeClientEviction too, PTAL.

Reply:
Nothing extra is being done here for parsing, just storing them as new lines. So might not be a need for unit testing.

Reply:
Everything that can be unit-tested, should ideally be unit-tested :smile:

Reply:
Yes, updated. PTAL 😄